ark/attest/__tests__/completions.test.ts

```diff
@@ -0,0 +1,121 @@
+import { scope, type } from "arktype"
+import assert from "assert"
+import { describe, test } from "mocha"
+import { attest } from "../assert/attest.js"
+
+type Obj = {
+	prop1: string
+	prop2: string
+	extra: unknown
+}
+
+describe("autocompletes", () => {
+	describe("String completions", () => {
+		test("Allows different quote types", () => {
+			//@ts-expect-error
+			attest(() => type("st")).type.completions.snap({ st: ["string"] })
+		})
+		test("Value completion with type", () => {
+			//@ts-expect-error
+			attest(() => type({ a: "n", b: "b" })).type.completions.snap({
+				n: ["number", "null", "never"],
+				b: ["bigint", "boolean"]
+			})
+		})
+		test("Value completion with scope", () => {
+			//@ts-expect-error
+			attest(() => scope({ a: "st" })).type.completions.snap({ st: ["string"] })
+		})
+		test("Completions shortcut", () => {
+			//@ts-expect-error
+			attest(() => scope({ a: "u" })).completions.snap({
+				u: ["undefined", "unknown", "uppercase", "uuid", "url"]
+			})
+		})
+		test("Both formats provide exact results", () => {
+			//@ts-expect-error
+			attest(() => type({ a: "u" })).completions({
+				u: ["undefined", "unknown", "uppercase", "uuid", "url"]
+			})
+			//@ts-expect-error
+			attest(() => type({ a: "u" })).type.completions({
+				u: ["undefined", "unknown", "uppercase", "uuid", "url"]
+			})
+		})
+	})
+
+	describe("Key completions", () => {
+		test("Casted object", () => {
+			//@ts-expect-error
+			attest(() => type({ "": "" } as Obj)).completions.snap({
+				"": ["extra", "prop1", "prop2"]
+			})
+		})
+		test("Lists available keys", () => {
+			//@ts-expect-error
+			attest(() => type("str", "@", { "": "" })).completions.snap({
+				str: ["string"],
+				"": ["description"]
+			})
+		})
+	})
+
+	describe("Property access", () => {
+		test("Autocompletes properties", () => {
+			//@ts-expect-error
+			attest(() => type("string")[""]).type.completions.snap({
```

Comment:
We should probably stop testing everything on `type` since this is isolated from `arktype` and this would break if we ever add a new method to type 😝